pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,120 @@ package sdkstats
 
 import (
 	"context"
-	"fmt"
-	"strconv"
 	"strings"
 
 	"github.com/google/go-github/github"
 )
 
 //RepoMetadata structure for storing Repository details
 type RepoMetadata struct {
-	FullName     string
+	URL          string
+	Name         string
+	Owner        string
 	ProjectType  string
 	Stars        int
-	SDKVersion   int
+	SDKVersion   string
 	CreatedAt    string
 	PushedAt     string
 	TotalCommits int
 }
 
-//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
-func GetStats(client *github.Client, q string) ([]RepoMetadata, error) {
-	var repoList []RepoMetadata
-	var projectType string
-	var sdkVersion int
-
-	//Search function return CodeResults for the q queery, and also Basic repository data.
-	searchOp, err := Search(client, q)
-
-	if err != nil {
-		return nil, err
-	}
-
-	if strings.Contains(q, "helm") {
-		projectType = "helm"
-	}
-	if strings.Contains(q, "ansible") {
-		projectType = "ansible"
-	}
-	if strings.Contains(q, "go.mod") {
-		projectType = "go.mod"
-	}
-	if strings.Contains(q, "Gopkg.toml") {
-		projectType = "Gopkg.toml"
-	}
-	for j := 0; j < len(searchOp); j++ {
-		for i := 0; i < len(searchOp[j].CodeResults); i++ {
-			owner := searchOp[j].CodeResults[i].GetRepository().GetOwner().GetLogin()
-			name := searchOp[j].CodeResults[i].GetRepository().GetName()
-			fullName := searchOp[j].CodeResults[i].GetRepository().GetFullName()
-
-			//getVersion returns SDK version by doing Fragment search
-			if projectType == "Gopkg.toml" {
-				sdkVersion = 0
-			} else {
-				sdkVersion = getversion(searchOp[j].CodeResults[i], projectType)
-			}
+//RepoMetadataQuery struct to store Search queries, and Project Type
+type RepoMetadataQuery struct {
+	ProjectType   string
+	Queries       []string
+	VersionParser VersionParser
+}
 
-			//GetRepoDetails returns Repository specififc details
-			repoMap := GetRepoDetails(client, owner, name)
+//VersionParser Interface implements ParseVersion, to Parse Version number from a given Text Match result.
+type VersionParser interface {
+	ParseVersion(codeResults github.CodeResult, projectType string) (string, error)
+}
 
-			TotalCommits, err := strconv.Atoi(repoMap["TotalCommits"])
-			Stars, err := strconv.Atoi(repoMap["Stars"])
-			if err != nil {
-				return nil, err
-			}
+//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
+func GetStats(client *github.Client, rq RepoMetadataQuery) ([]RepoMetadata, error) {
```

Comment:
This is the only function used in `main.go`, so that's probably a good indication that the other functions  (e.g. `Search` and `GetRepoDetails`) do not need to be exported and made public. To unexport the other functions, simply make the the first letter of their name lowercase.

An extremely important aspect of maintaining libraries is to be very careful and deliberate about designing and changing their public API surface. Any breaking changes to a library make life more difficult for users of the library, so it's best to keep the API surface as small and intuitive as possible.

Later versions of libraries can always add new functionality and expose more parts of the API surface without breaking existing users, but going the opposite direction and making already public APIs private is a breaking change.

Reply:
The reason I made these exportable, is that, In case other users have the need to let say only call "Search()", or "GetRepoDetails", It would be simple enough to call them. WDYT. This also makes case, NOT to pass "CodeResult" to GetRepoDetails(). WDYT?

Reply:
I think You mahe missed reviewing this comment. I was thinking if we can make GetRepoDetails() exportable as well, along with "Search()". WDYT

Reply:
Will make them unexportable based on [this comment](https://github.com/learnoperators/go-github-metrics/pull/12#discussion_r341430381)

Reply:
See my comment [here](https://github.com/learnoperators/go-github-metrics/pull/12/files/e8c3e35e732f5354b822740fd70472a8565c4f01#r341430381)